source/clear-linux/tutorials/hadoop.rst

```diff
@@ -0,0 +1,255 @@
+
+   .. code-block:: bash
+
+      sudo mkdir /etc/hadoop
+
+#. Copy the defaults from :file:`/usr/share/defaults/hadoop` to
+   :file:`/etc/hadoop` with the following command:
+
+   .. code-block:: bash
+
+      $ sudo cp /usr/share/defaults/hadoop/* /etc/hadoop
+
+   .. note:: Since |CL| is a stateless system, you should never modify the
+      files under the :file:`/usr/share/defaults` directory. The software
+      updater will overwrite those files.
+
+Once all configuration files are in :file:`/etc/hadoop`, we must edit them
+according to our needs. The first file, :file:`/etc/hadoop/core-site.xml`, is
+responsible to inform the Hadoop daemon of where `NameNode` is running. The
+`NameNode` server is the master server managing the files system namespace
+and regulating the clients' access to files.
+
+In this tutorial, our `NameNode` runs in our `localhost`. Follow these steps
+to set it up correctly:
+
+#. Open the :file:`/etc/hadoop/core-site.xml` using the editor of your
+   choice and modify the file as follows:
+
+   .. code-block:: xml
+
+      <?xml version="1.0" encoding="UTF-8"?>
+      <?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+      <configuration>
+      <property>
+      <name>fs.default.name</name>
+      <value>hdfs://localhost:9000</value>
+      </property>
+      </configuration>
+
+#. Edit :file:`/etc/hadoop/hdfs-site.xml`. This file configures the
+   :abbr:`HDFS (Hadoop Distributed File System)` daemons. This configuration
+   includes the list of permitted and excluded data nodes and the size of
+   said blocks. In this example, we are setting the number of block
+   replication to 1 from the default of 3 as follows:
+
+   .. code-block:: xml
+      :emphasize-lines: 6
+
+      <?xml version="1.0" encoding="UTF-8"?>
+      <?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+      <configuration>
+      <property>
+      <name>dfs.replication</name>
+      <value>1</value>
+      </property>
+      <property>
+      <name>dfs.permission</name>
+      <value>false</value>
+      </property>
+      </configuration>
+
+#. Edit :file:`/etc/hadoop/mapred-site.xml`. This file configures all daemons
+   related to MapReduce: `JobTracker` and `TaskTrackers`. With MapReduce,
+   Hadoop can process big amounts of data in multiple systems. In our
+   example, we set :abbr:`YARN (Yet Another Resource Manager)` as our runtime
+   framework for executing MapReduce jobs as follows:
+
+   .. code-block:: xml
+      :emphasize-lines: 5,6
+
+      <?xml version="1.0" encoding="UTF-8"?>
+      <?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+      <configuration>
+      <property>
+      <name>mapreduce.framework.name</name>
+      <value>yarn</value>
+      </property>
+      </configuration>
+
+#. Edit :file:`/etc/hadoop/yarn-site.xml`. This file configures all daemons
+   related to YARN: `ResourceManager` and `NodeManager`. In our example, we
```

Comment:
State what YARN is and use literal text

Reply:
The YARN abbreviation was stated above. Similarly to MapReduce, I don't think that we should be explaining what all these pieces are here. IMO it is enough to state that YARN is the runtime framework being used, and that much has been stated above.

Reply:
Understood and fine by me! :)